src/coord/src/coord.rs

```diff
@@ -2393,7 +2393,7 @@ pub fn auto_generate_src_idx(
 ) -> catalog::Index {
     auto_generate_primary_idx(
```

Comment:
Hmm given that `CatalogItem::primary_idx_keys exists` now we could consider getting rid of `auto_generate_src_idx` and `auto_generate_view_idx`.

Which would also have the benefit that the rules about what becomes the keys for a view vs source is only in one place instead of two.